bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -875,6 +881,20 @@ public void run() {
     @Override
     public void modifiedService(ServiceReference/* <ModuleHandlerFactory> */ reference,
             /* ModuleHandlerFactory */Object service) {
+        logger.debug("ModuleHandlerFactory modified, updating handlers");
+        ModuleHandlerFactory moduleHandlerFactory = ((ModuleHandlerFactory) service);
+        @SuppressWarnings("unchecked")
+        Collection<String> removed_handlers = (Collection<String>) reference.getProperty("removed_handlers");
```

Comment:
Does this mean that the factory has to always correctly set these properties?
Can we find an implementation that can do without that requirement?
Since we have the moduleHandlerFactories map, couldn't we iterate through it and remove all module types that are not anymore in the "new" list of supported types and simply add all types that are not yet in the moduleHandlerFactories map?